fix: nested scope filter

jannyHou opened the pull request:
Fixes https://github.com/strongloop/loopback-next/issues/5872

Nested inclusion filter like 

```ts
{
  include: [{
    relation: 'todos',
    scope: {
      // after enabling coercion against spec schema in PR 
      // https://github.com/strongloop/loopback-next/pull/5808
      // it breaks cuz scope schema doesn't have property include.
      include: [{
        relation: 'someOtherRelation',
      }]
    }
  }]
}
```

Fails. Because the `scope` schema is generated from `EmptyModel`, which doesn't have a relation:

https://github.com/strongloop/loopback-next/blob/683e162de982acde34b45101383a5d69cf1c0df1/packages/repository-json-schema/src/filter-json-schema.ts#L38

and thus does not have `include` generated in its `properties`:

https://github.com/strongloop/loopback-next/blob/683e162de982acde34b45101383a5d69cf1c0df1/packages/repository-json-schema/src/filter-json-schema.ts#L118-L120

This PR adds an any inclusion schema for the empty model's scope filter schema.


## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

agnes512 commented:
Just wondering what was the reason it `excluded scope.include`?

jannyHou commented:
@agnes512 good question, if you look at the last screenshot, 

`if (hasRelations && !excluded.includes('include'))` decides not add the include property because `EmptyModel` DOES NOT have relations.

agnes512 commented:
Will the scope be coerced? If so, can we have a test for it? Not sure if this is a valid case :p

jannyHou commented:
@agnes512 
> Will the scope be coerced? 

No it won't be coerced, it's an any object...

agnes512 commented:
> No it won't be coerced, it's an any object...

Does that mean for filters like 
```
/orders?filter[include][0][relation]=shipments&filter[include][0][scope]filter[fields][name]=false
```
would not have the expected result? (`fields` inside of the scope will be a string `false`)

jannyHou commented:
@agnes512 Right the false will be a string. The scope filter schema is generated from `EmptyModel` which doesn't have the ability to generate fields schema according to `TodoList` model. I verified

> http://127.0.0.1:3000/todo-lists?filter[include][0][relation]=todos&filter[include][0][scope][fields][desc]=false

returns wrong result. But the corresponding API flavor call returns correctly.

hacksparrow commented:
> The scope filter schema is generated from EmptyModel which doesn't have the ability to generate fields schema according to TodoList model. I verified.

Let's do the verification with a test.

agnes512 commented:
>  I verified nhttp://127.0.0.1:3000/todo-lists?filter[include][0][relation]=todos&filter[include][0][scope][fields][desc]=false
returns wrong 

It seems a limitation to me. Can we also document it?

jannyHou commented:
cc @bajtos @hacksparrow Thanks for review 👍 , added tests in https://github.com/strongloop/loopback-next/pull/5908/commits/fa097ea8810ff24074a332e11ba1744096b94524

cc @agnes512 
> Can we also document it?

Sure I added notes in the test case, I will mention it in doc as well, next commit coming soon.
